File: app/backend/util/config/ByodConfigHandler.java
package backend.util.config;

import java.util.Optional;

import play.mvc.Http;
import play.mvc.Result;

public interface ByodConfigHandler {

    byte[] getExamConfig(String hash, byte[] pwd, String salt);
    String calculateConfigKey(String hash);
    String getPlaintextPassword(byte[] pwd, String salt);
    byte[] getEncryptedPassword(String pwd, String salt);
    Optional<Result> checkUserAgent(Http.RequestHeader request, String examConfigKey);

}

File: app/backend/util/config/ByodConfigHandler.scala
package backend.util.config

import java.util.Optional

import play.mvc.{Http, Result}

trait ByodConfigHandler {
  def getExamConfig(hash: String, pwd: Array[Byte], salt: String): Array[Byte]
  def calculateConfigKey(hash: String): String
  def getPlaintextPassword(pwd: Array[Byte], salt: String): String
  def getEncryptedPassword(pwd: String, salt: String): Array[Byte]
  def checkUserAgent(request: Http.RequestHeader, examConfigKey: String): Optional[Result]
}
